python/flatbuffers/builder.py

```diff
@@ -518,13 +518,20 @@ def Slot(self, slotnum):
         self.current_vtable[slotnum] = self.Offset()
     ## @endcond
 
-    def __Finish(self, rootTable, sizePrefix):
+    def __Finish(self, rootTable, sizePrefix, file_identifier=None):
         """Finish finalizes a buffer, pointing to the given `rootTable`."""
         N.enforce_number(rootTable, N.UOffsetTFlags)
         prepSize = N.UOffsetTFlags.bytewidth
         if sizePrefix:
             prepSize += N.Int32Flags.bytewidth
+        if(file_identifier is not None):
```

Comment:
don't think Python needs `()` inside `if`.

Reply:
> 
> 
> the schema compiler will currently accept any bytes for `file_identifier` (since it is a string constant where escape characters are permitted), but the intent was for it to only be printable ascii characters (32..126).
> 
> The reason being that when such an identifier is missing, the memory being tested may contain the start of a vtable (a 16bit vtable size & object size). With the smallest ascii characters, you'd need a vtable size >= 0x2020 (> 4000 fields) before you run the risk that `BufferHasIdentifier` would accidentally return true for the wrong kind of buffer. This is still an issue, I guess we could say that the use of the `file_identifier` feature is not safe for users of really big tables. This is bad design, caused by adding this feature to the format retroactively. It should not bite anyone in practice though.
> 
> So in summary, I'd be in favor to restricting `file_identifier` to printable ascii where possible. I'd even be in favor in retro-actively adding this to `flatc`, maybe at first as a deprecation warning, just in case anyone actually uses this.

In new versions of flatc, you could add a default file_identifier of `\0\0\0\0`, which, by also only allowing printable ASCII characters, would be completely unambiguous.

Reply:
I made the change